pkg/api/validation/validation.go

```diff
@@ -1278,6 +1291,18 @@ func ValidateSecret(secret *api.Secret) errs.ValidationErrorList {
 		}
 	case api.SecretTypeOpaque, "":
 		// no-op
+	case api.SecretTypeDockercfg:
+		dockercfgBytes, exists := secret.Data[api.DockerConfigKey]
+		if !exists {
+			allErrs = append(allErrs, errs.NewFieldRequired(fmt.Sprintf("data[%s]", api.DockerConfigKey)))
+			break
+		}
+
+		dockercfg := &credentialprovider.DockerConfig{}
+		if err := json.Unmarshal(dockercfgBytes, dockercfg); err != nil {
```

Comment:
> there's no good way to test a dockercfg

We can identify things we know won't work because they're not well formed. We can't validate the credentials all the way to the destinations registry (nor should we here)

> there's no real way to validate this

Think of it like format validation. Do we validate the format of the `image` field in container definitions? If so, I'd consider this analogous. If we accept any value for the `image` field and let bogus things just fail on the kubelet, I guess we can do the same here.